openquake/commonlib/readinput.py

```diff
@@ -485,14 +485,15 @@ def __call__(self, fname, sm, apply_uncertainties, investigation_time):
         :param investigation_time:
             the investigation_time in the job.ini
         :returns:
-            a copy of the original source model with possibly changed sources
+            a copy of the original source model with changed sources (if any)
+            or the original model with unchanged sources
         """
-        sm = copy.deepcopy(sm)
         check_nonparametric_sources(fname, sm, investigation_time)
-        for group in sm:
-            for src in group:
-                changed = apply_uncertainties(src)
-                if changed:
+        if apply_uncertainties:
```

Comment:
Not sure what are you checking here. Does it means that the argument that is passed to the method can be 'False' ? If yes please update the docstring.